text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -0,0 +1,1505 @@
+                                                           best_performance_update_idx])),
+                                ('elapsed_time', int(time.time() - start_tick))]
+                if eval_metric_scorer._sign < 0:
+                    report_items.append(('reward_attr', -dev_score))
+                else:
+                    report_items.append(('reward_attr', dev_score))
+                report_items.append(('eval_metric', eval_metric_scorer.name))
+                report_items.append(('exp_dir', exp_dir))
+                if find_better:
+                    best_report_items = report_items
+                reporter(**dict(report_items))
+                report_local_jsonl_f.write(json.dumps(dict(report_items)) + '\n')
+                report_local_jsonl_f.flush()
+                report_idx += 1
+            if no_better_rounds >= cfg.optimization.early_stopping_patience:
+                logger.info('Early stopping patience reached!')
+                break
+            total_time_spent = time.time() - start_tick
+            if time_limit is not None and total_time_spent > time_limit:
+                break
+    # Average checkpoints
+    best_report_items_dict = dict(best_report_items)
+    best_report_items_dict['report_idx'] = report_idx + 1
+    reporter(**best_report_items_dict)
+    report_local_jsonl_f.write(json.dumps(best_report_items_dict) + '\n')
+    report_local_jsonl_f.close()
+
+
+def get_recommended_resource(nthreads_per_trial=None,
+                             ngpus_per_trial=None) -> Tuple[int, int]:
+    """Get the recommended resources.
+
+    Internally, we will try to use GPU whenever it's possible. That means, we will use
+    a single GPU for finetuning.
+
+    Parameters
+    ----------
+    nthreads_per_trial
+        The number of threads per trial provided by the user.
+    ngpus_per_trial
+        The number of GPUs per trial provided by the user.
+
+    Returns
+    -------
+    nthreads_per_trial
+        The recommended resource.
+    ngpus_per_trial
+    """
+    if nthreads_per_trial is None and ngpus_per_trial is None:
+        nthreads_per_trial = get_cpu_count()
+        ngpus_per_trial = 1
+    elif nthreads_per_trial is not None and ngpus_per_trial is None:
+        ngpus_per_trial = 1
+    elif nthreads_per_trial is None and ngpus_per_trial is not None:
+        if ngpus_per_trial != 0:
+            num_parallel_jobs = get_gpu_count() // ngpus_per_trial
+            nthreads_per_trial = max(get_cpu_count() // num_parallel_jobs, 1)
+        else:
+            nthreads_per_trial = get_cpu_count()
+    nthreads_per_trial = min(nthreads_per_trial, get_cpu_count())
+    ngpus_per_trial = min(ngpus_per_trial, get_gpu_count())
+    assert nthreads_per_trial > 0 and ngpus_per_trial >= 0,\
+        'Invalid number of threads and number of GPUs.'
+    return nthreads_per_trial, ngpus_per_trial
+
+
+@use_np
+class MultiModalTextModel:
```

Comment:
Is this a public API? If so, rename various things to align closer to predictors, such as:

output_directory -> path, dir_path -> path, test_data -> data

Reply:
This should be mostly private I think.